packages/fuels-test-helpers/src/utils.rs

```diff
@@ -14,3 +18,180 @@ pub fn into_message_configs(messages: Vec<Message>) -> Vec<MessageConfig> {
+
+    let mut crate_info_response = match isahc::get(crate_info_url) {
+        Ok(response) => response,
+        Err(err) => {
+            eprintln!("Failed to fetch dependency information: {}", err);
+            return;
+        }
+    };
+
+    let crate_info_body = match crate_info_response.text() {
+        Ok(body) => body,
+        Err(err) => {
+            eprintln!("Failed to fetch dependency information: {}", err);
+            return;
+        }
+    };
+
+    let crate_info_json: Result<Value, serde_json::error::Error> =
+        match serde_json::from_str(&crate_info_body) {
+            Ok(json) => Ok(json),
+            Err(err) => {
+                eprintln!("Failed to fetch dependency information: {}", err);
+                return;
+            }
+        };
+
+    let latest_version = match crate_info_json {
+        Ok(ref json) => match json["versions"][0]["num"].as_str() {
+            Some(version) => version,
+            None => {
+                eprintln!("Failed to fetch dependency information");
+                return;
+            }
+        },
+        Err(err) => {
+            eprintln!("Failed to fetch dependency information: {}", err);
+            return;
+        }
+    };
+
+    let dependency_info_url = format!(
+        "https://crates.io/api/v1/crates/{}/{}/dependencies",
+        crate_name, latest_version
+    );
+
+    let mut dependency_info_response = match isahc::get(dependency_info_url) {
+        Ok(response) => response,
+        Err(err) => {
+            eprintln!("Failed to fetch dependency information: {}", err);
+            return;
+        }
+    };
+
+    let dependency_info_body = match dependency_info_response.text() {
+        Ok(body) => body,
+        Err(err) => {
+            eprintln!("Failed to fetch dependency information: {}", err);
+            return;
+        }
+    };
+
+    let dependency_info_json: Result<Value, serde_json::error::Error> =
+        serde_json::from_str(&dependency_info_body);
+
+    if let Ok(json) = dependency_info_json {
+        if let Some(dependencies) = json["dependencies"].as_array() {
+            if let Some(fuel_core_dependency) = dependencies
+                .iter()
+                .find(|dependency| dependency["crate_id"].as_str() == Some("fuel-core"))
+            {
+                let fuel_core_req_vec = match fuel_core_dependency["req"].as_str() {
+                    Some(req) => req.trim_start_matches('^').to_string(),
+                    None => {
+                        eprintln!("Failed to retrieve fuel-core requirement");
+                        return;
+                    }
+                };
+
+                let fuel_core_req = fuel_core_req_vec
+                    .split('.')
+                    .take(2)
+                    .collect::<Vec<&str>>()
+                    .join(".");
+
+                if fuel_core_req != used_fuel_version {
+                    eprintln!(
```

Comment:
We shouldn't spam `stderr`. Any message you wish to relay is best done through logging at the appropriate level.